pwndbg/commands/windbg.py

```diff
@@ -8,17 +8,28 @@
 import codecs
 from itertools import chain
 
-import gdb
-
+import pwndbg.aglib.arch
+import pwndbg.aglib.memory
+import pwndbg.aglib.strings
+import pwndbg.aglib.typeinfo
 import pwndbg.commands
-import pwndbg.gdblib.arch
-import pwndbg.gdblib.memory
-import pwndbg.gdblib.strings
-import pwndbg.gdblib.symbol
-import pwndbg.gdblib.typeinfo
-import pwndbg.hexdump
 from pwndbg.commands import CommandCategory
 
+if pwndbg.dbg.is_gdblib_available():
+    import gdb
+
+
+def enhex(size, value):
+    value = value & ((1 << 8 * size) - 1)
+    x = "%x" % abs(value)
+    x = x.rjust(size * 2, "0")
+    return x
+
+
+# `pwndbg.hexdump` imports `enhex` from this module, so we have to import it
+# after it's been defined in order to avoid circular import errors.
+import pwndbg.hexdump
```

Comment:
I don't quite understand this. It seems still to be a circular import and it should be the same/similar imports before. Why is it an issue now and wasn't before?

Reply:
> It seems still to be a circular import and it should be the same/similar imports before.

It still is, but, to the best of my knowledge, by importing it after `enhex` has been defined, we allow [the module code execution step of the import](https://docs.python.org/3/reference/import.html#loading) to assign a value to `enhex` before the circular import happens, and thus, `pwndbg.hexdump` always sees it defined. The issue I was getting before I moved the circular import statement down was that `enhex` wasn't defined in `pwndbg.commands.windbg`, which was correct, because it very much hadn't been by that point in execution order.

> Why is it an issue now and wasn't before?

I can't say for sure, because I didn't actually spend any time digging into what import specifically caused it to break. But the thing is, circular imports are very sensitive to the order in which you first import things, and the LLDB port refactors shuffled around the order of many of the import statements in Pwndbg. My best guess is that the original relative import order for `pwndbg.commands.windbg` and `pwndbg.hexdump` got reversed somewhere along the port, and that's what caused it to break.

Reply:
I think the original order might've had `pwndbg.hexdump` come first, and `pwndbg.commands.windbg` come later. That would keep the `import pwndbg.hexdump` statement in `pwndbg.commands.windbg` from triggering the code execution step for `pwndbg.hexdump` before it had defined `enhex`, which is what caused our ciruclar import error.

Moving the import down also means either relative import order works now.